### ISSUE TITLE
zone identifier maldocs?

### ISSUE THREAD

**Issue by mallorybobalice** [spender-sandbox/cuckoo-modified#268]
hi, 

anyone seeing maldoc samples (say C:\Users\xxxx\AppData\Local\Temp\yyy.doc) querying for  C:\Users\xxxx\AppData\Local\Temp\yyy.doc:Zone.Identifier  ADS via NtQueryAttributesFile (well via FileSystemObject FileExists
and not executing correctly if not found? (the samples i have seem to only check whether it exists not the content) 

Anything we can do - for example fake say it exists via cuckoomon, or create one via disguise.py?

**Comment by doomedraven** [comment]
yes is self protection, if zone identifier presented it was downloaded from internet, if not, was copied

**Comment by mallorybobalice** [comment]
Well, I do understand ie and possibly outlook and chrome make it when file is saved from the internet zone. 
I was hoping someone could add making the samplepath:Zone.Identifier
[ZoneTransfer] 
ZoneId=3

Via disguise. Py similar to what was done for office recent nums

**Comment by mallorybobalice** [comment]
=) anyone https://github.com/spender-sandbox/cuckoo-modified/blob/master/analyzer/windows/modules/auxiliary/disguise.py ?

**Comment by mallorybobalice** [comment]
Tho I guess it's easier to add https://github.com/spender-sandbox/cuckoo-modified/blob/master/agent/agent.py 
After sample.write(data)

**Comment by spender-sandbox** [comment]
It can be done via analyzer.py, on 'target'.  That wouldn't require any VM updates.  Regardless, is this still be used?  It seems like they stopped using it once they realized it would affect being able to package the files in zips, etc.  Which would be a problem with doing it blindly on every submission, because the same check could be performed in inverse where it's known that ADS shouldn't exist.

-Brad

**Comment by mallorybobalice** [comment]
Was seeing it in mal docs for about 3 weeks (including this week ). 

For zips, aren't they demuxed into individual files after going to csb? (Or only for some submission methods? )
I haven't seen many zipped docs(though I do occasionally resubmitted batches as zips via the web ui where it unzips (demuxes? )them to individual submissions ).of course other people may see a few 

I think if files downloaded via browsers or saved from outlook make the ADS we should add it to csb to mimic the OS behavior (assuming ntfs and most interesting files coming from the internet zone. )However that sounds like a rather debatable point. So perhaps it can have a toggle flag on submit.py, web and api

I'm not sure what a good way of adding it would be or if things other than maldocs use this for evasion or how to handle compressed docs and batch submission to demux .  Perhaps it can be added as office feature and we could pass back a flag if it's submitted as part of a zip with an override option to still add the ADS for the individual demuxed docs?(if only ui demixes)

Hmm yea you're right if done via agent.py would require guest updates (annoying ).re analyzer.py, target, not quite sure, bit more hints please? 

Guess we have the full path here 
```
self.target = os.path.join(os.environ["TEMP"] + os.sep,                                       str(self.config.file_name))
```
And could append the :zoneinfo to the path and then create the stream and might as well populate it with the right info? (As in keep it around the original lines were populating target?

**Comment by mallorybobalice** [comment]
Ps is there a signature for read ads?

**Comment by mallorybobalice** [comment]
hmmm, i've had a look after being away for a while, and we're still getting 2-3 campaigns checking for zone info per week.... you guys interested in including it into CSB itself at all?

**Comment by mallorybobalice** [comment]
hmmm, actually I'm wrong. maybe these guys did move from ADS

after digging in the VBA
recent docs 3 - old
doc  name is the original name  - new yea that's annoying 
window titles - e.g. -wireshark process explorer Visual basic office debug window (also annoying)
enough windows or processes  # great...gotta update the guests with some benign background processes 
maxmind -old

**Comment by doomedraven** [comment]
@mallorybobalice you can patch all this from host, patching the .doc package ;)

**Comment by mallorybobalice** [comment]
>@mallorybobalice you can patch all this from host, patching the .doc package ;)


as in for analysis - run winword, and also run 10x benign process that doesn't do much  stuff   (calc, whatever) ? 

well, can be excel or word but hmm... 

thing is - some of these are generic detections and I kinda think I should update the child VMs regardless of package.

**Comment by doomedraven** [comment]
you have a lot of default soft to run ;)

**Comment by parker67** [comment]
So when we first discovered this we chatted about how we would implement it into our current cuckoo. 
I think it would be best to actually have it as an optional in the submission page to give the option to generate the zone.identifier or not.
Just my 2c

**Comment by mallorybobalice** [comment]
I think it should be both a ui option and a default zone in the config for say cuckoo. Conf.and \ Or perhaps passable parameter via the api and submit.Py
That way it's explicitly controlled not implied as files "not coming from the net ". If I'm not mistaken both outlook and ie tag it. I'm unclear if 7z and win shell unzip preserve the zone identifier info

**Comment by mallorybobalice** [comment]
Possibly chrome too 

Like always I'm happy to be wrong but could we please discuss this 
Could even be granular per target analysis file type 

@spender-sandbox
